mam/v2/wots/wots.c

```diff
@@ -154,15 +152,34 @@ void wots_recover(spongos_t *const spongos, trits_t const hash,
 
   trits_copy(signature, sig_pks);
   wots_hash_sign_or_recover(spongos, sig_pks, hash, WOTS_HASH_RECOVER);
-  spongos_hash(spongos, sig_pks, public_key);
+  mam_spongos_hash(spongos, sig_pks, public_key);
 }
 
-bool wots_verify(spongos_t *const spongos, trits_t const hash,
-                 trits_t const signature, trits_t const public_key) {
+bool mam_wots_verify(mam_spongos_t *const spongos, trits_t const hash,
+                     trits_t const signature, trits_t const public_key) {
   MAM2_TRITS_DEF0(recovered_public_key, MAM2_WOTS_PK_SIZE);
   recovered_public_key =
       MAM2_TRITS_INIT(recovered_public_key, MAM2_WOTS_PK_SIZE);
 
-  wots_recover(spongos, hash, signature, recovered_public_key);
+  mam_wots_recover(spongos, hash, signature, recovered_public_key);
   return trits_cmp_eq(public_key, recovered_public_key);
 }
+
+size_t mam_wots_serialized_size(mam_wots_t const *const wots) {
+  return mam_spongos_serialized_size(&wots->spongos) +
+         pb3_sizeof_ntrytes(MAM2_WOTS_SK_SIZE);
+}
+
+void mam_wots_serialize(mam_wots_t const *const wots, trits_t trits) {
```

Comment:
You need to take a pointer to `trits_t` so that PB3 operations can advance the pointer, then the drop becomes useless